src/Containers/Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -8,26 +8,29 @@
 using System.Net;
 using System.Net.Http.Headers;
 using System.Net.Http.Json;
+using System.Security.Cryptography;
 using System.Text;
 using System.Text.Json;
 using System.Text.Json.Nodes;
 
 namespace Microsoft.NET.Build.Containers;
 
+public enum ProgressMessageLevel {
+    Info,
+    Trace
+}
+
+public record struct LogMessage (ProgressMessageLevel level, string messageFormat, object[] formatArgs) {
+    public static LogMessage Info(string messageFormat, params object[] formatArgs) => new LogMessage(ProgressMessageLevel.Info, messageFormat,  formatArgs);
+    public static LogMessage Trace(string messageFormat, params object[] formatArgs) => new LogMessage(ProgressMessageLevel.Trace, messageFormat,  formatArgs);
+};
```

Comment:
Finally, a note about logs - I added trace-level logs for the low-level details that I don't think users need to see, but that can be useful when debugging what's gone on.  I also needed a way to expose a 'level' for these logs, so that we could map that onto MSBuild Information severity (either High or Low). Eventually some kind of Microsoft.Extensions.Logging layer might be useful here to prevent this kind of yet-another-abstraction, but it was simple to do here.